Make radios and checkboxes components easier to link to from error summary (#1426)
alphagov/govuk-frontend: CHANGELOG.md

💥 Breaking changes:

- Make components easier to link to from error summary

  Updates the radios, checkboxes and date input components so it's easier to link to their first items from the error summary component.

  To migrate:

>   You will need to ensure that your error summary still links correctly to your first input by removing any logic that adds a suffix to your id.

Author: I don't think this is very good, would welcome some ideas to make this better. Potentially some before and after examples?

💥 Breaking changes:

- Make radios and checkboxes components easier to link to from the error summary component

>   The first input in the radios and checkboxes will have an `id` and `name` attribute that match, allowing for easier generation of error summary links.

Reviewer: This is only true if you haven't specified your own `idPrefix`, I think?

Reviewer: I wonder if we should be more explicit about what's actually changed here?

Perhaps something like…

> The ID of the first input in a set of radios or checkboxes will no longer be suffixed with `-1`. This means that it will be the same as your `idPrefix`, or if `idPrefix` is not set it'll be the same as the `name`
>
> We've made this change to make it easier to link from the error summary to radios or checkboxes.

>   You should check that clicking an error link for a radios or checkboxes results in the first input being focused.

Reviewer: Is it clear enough what we mean by 'error link'? Would it be better to use something like:

```suggestion
  You should check that clicking a link from the error summary to radios or checkboxes results in the first input being focused.
```

>   To migrate you can update your error summary so it links to the first element, for example if you are using an `idPrefix` option of `question`, your error summary can now link to `#question` instead of `#question-1`.

Reviewer: Users might also be overriding the `id` of the first radio/checkbox, which I think is what we suggested in #929. In that case, they could remove the `id` as it won't be doing anything.

Author: I figured since this wasnt required that I'd leave the up to people themselves rather than potentially confusing things... maybe should just have this in

Author: I tried this and I'm worried that there's just a lot of steps being described that could confuse people so I wanted to focus on the things they definitely need to do...

Author: nevermind I've included it after all..

  If you cannot change your error summary, you could change your component to set the `item.id` option for the first input, for example:

  ```javascript

>   {{ govukRadios({

Reviewer: This example is missing a `name` attribute, which is required.
